app/models/agents/rss_agent.rb

```diff
@@ -170,6 +170,20 @@ def check_and_track(entry_id)
       require 'feedjira_extension'
     end
 
+    def preprocessed_body(response)
+      body = response.body
+      case body.encoding
+      when Encoding::ASCII_8BIT
+        # Encoding is unknown from the Content-Type, so let the SAX
+        # parser detect it from the content.
+      else
+        # Encoding is already known, so do not let the parser detect
+        # it from the XML declaration in the content.
+        body.sub!(/(<\?xml(?:\s+\w+\s*=\s*(['"]).*?\2)*)\s+encoding\s*=\s*(['"]).*?\3/, '\\1')
```

Comment:
Yeah, I guess so.  The specification says there should be nothing before an XML declaration, but it'd be safer to allow leading spaces.  There could be also a possibility for a document to have a BOM.